src/ui/molecules/navigation/navigation.chain.dropdown.js

```diff
@@ -25,11 +32,30 @@ const chains = [
 ];
 
 export default function NavigationChainDropdown({ chainId, setChainId }) {
+  const { width } = useWindowDimensions();
+
+  const chainToIcon = (chain) => {
+    switch (chain.toLowerCase()) {
+      case "mainnet":
+        return <ETH className="h-5 w-5" />;
+      case "arbitrum":
+        return <ARB className="h-5 w-5" />;
+      case "polygon":
+        return <MATIC className="h-5 w-5" />;
+      case "avalanche":
+        return <AVAX className="h-5 w-5" />;
+    }
+  };
+
   return (
     <Menu as="div" className="relative inline-block text-left">
       <div>
         <Menu.Button className="inline-flex font-bold rounded-lg justify-center w-full bg-zero-green-500/90 hover:bg-zero-green-500/40 shadow-sm px-2 py-1 text-sm md:text-base text-badger-white-400 focus:outline-none ">
-          {chainId ? getChainName(chainId) : "Loading..."}
+          {chainId
+            ? width > 600
+              ? getChainName(chainId)
+              : chainToIcon(getChainName(chainId))
```

Comment:
The default case for `getChainName` has been changed to return "Unsupported Chain," so we just need to handle that in chainIdToIcon as well now.

Reply:
Added default question mark